File: Python/problem0645.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        # nums.sort()
        # for k in range(1, len(nums)):
        #     if nums[k] == nums[k-1]:
        #         # tmp = nums.copy()
        #         return [ nums[k], list( set(range(1, len(nums)+1)) - set(nums) )[0] ]

        d = {}
        for num in nums:
            try:
                d[num] += 1
            except:
                d[num] = 1
        for k, v in d.items():
            if v == 2:
                break
        return [ k, list( set(range(1, len(nums)+1)) - set(nums) )[0] ]

File: C++/problem0645.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        map<int, int> m;
        for (int i = 0; i < nums.size(); ++i)
        {
            m.insert(pair<int, int>(i + 1, 0));
        }

        for (vector<int>::iterator it = nums.begin(); it < nums.end(); ++it)
        {
            m[*it]++;
        }

        vector<int> ans;
        for (map<int, int>::iterator it = m.begin(); it != m.end(); ++it)
        {
            if (it->second == 2)
            {
                ans.push_back(it->first);
            }
        }

        for (map<int, int>::iterator it = m.begin(); it != m.end(); ++it)
        {
            if (it->second == 0)
            {
                ans.push_back(it->first);
            }
        }

        return ans;

    }
};
